Patch for chest

Kitteh6660 opened the pull request:
Should fix the crash when attempting to load modded save in 1.4.15+.

Tested with 1.4.17 save and it should now work. Saving works as well.

Compatibility should be retained.

Kitteh6660 committed:
Patch for chest

tmedwards commented:
Issues:

1. You should be caching the result of `GetObj()`, rather than calling it repeatedly.
2. That's not the only instance of `GetObj("itemStorage")`.  It's also called on line ~180.  I suspect that if you've modified where it's homed, within `inventory`, then you should also update that call site as well.

That being the case, this would probably be better as something like:

```
        AmfObject GetItemStorageObj()
        {
            var itemStorage = GetObj("itemStorage");
            return itemStorage != null ? itemStorage : GetObj("inventory").GetObj("itemStorage");
        }

        void UpdateChest()
        {
            _chest.Clear();
            foreach (var pair in GetItemStorageObj()) _chest.Add(pair.ValueAsObject);
        }
```

Don't forget to update the `GetObj("itemStorage")` call on line ~180 to `GetItemStorageObj()`.

Kitteh6660 committed:
Update to XML File for Revamp 1.4.17

There may be a few missing entries but I've accounted for new items, flags, and body parts.

Kitteh6660 commented:
I will get that done! However, the refactor also changed the inventory so I will need to get that patched too. It no longer uses itemSlot1, itemSlot2, etc.

Kitteh6660 committed:
Updated chest according to suggestion

Kitteh6660 committed:
Merge branch 'patch-1'

papajon1212 commented:
hey want to play sometime ?
